packages/sitecore-jss-dev-tools/src/templating/nextjs/component-props.loader.ts

```diff
@@ -0,0 +1,77 @@
+import * as recast from 'recast';
+
+type VariableDeclaration = recast.types.namedTypes.VariableDeclaration;
+
+/**
+ * Webpack loader to strip functions from the source code
+ * Strips the `getServerSideProps` and `getStaticProps` functions from the source code
+ * @param {string} source file source code
+ * @returns {string} output file source code with stripped functions
+ */
+export default function componentPropsLoader(source: string) {
+  // Parse the source code into an AST (Abstract Syntax Tree)
+  const ast = recast.parse(source, {
+    parser: require('recast/parsers/babel-ts'),
+  });
+
+  // List of functions to strip from the AST
+  const functionsToStrip = ['getServerSideProps', 'getStaticProps'];
+
+  // Traverse the AST and strip the functions
+  recast.visit(ast, {
+    // Visit the named export function expression
+    visitExportNamedDeclaration: function(path): boolean | void {
+      // Get the variable declaration from the AST
+      (path.node.declaration as VariableDeclaration)?.declarations?.forEach((declaration) => {
+        // Check if the function is in the list of functions to strip
+        if (
+          'id' in declaration &&
+          'name' in declaration.id &&
+          typeof declaration.id.name === 'string' &&
+          functionsToStrip.includes(declaration.id.name)
+        ) {
+          // Strip the function from the AST
+          path.prune();
+
+          // Remove the function from the list of functions to strip
+          functionsToStrip.splice(functionsToStrip.indexOf(declaration.id.name), 1);
+        }
+      });
+
+      if (functionsToStrip.length === 0) {
+        // We have pruned all the functions we need to, so we can stop traversing the AST
+        return false;
+      }
+
+      // Continue traversing the AST
+      this.traverse(path);
+    },
+    // Visit the named export function declaration
+    visitFunctionDeclaration: function(path): boolean | void {
```

Comment:
I'm sure you already thought about reducing the duplication between these 2 "visit" functions, so assuming this is about as far as it makes sense to take it :)